### PR TITLE
Updates to avoid using FEProblem parameters to pass data

### PR DESCRIPTION
**Pull request by aeslaughter**
This implements a method to report values from and Executioner via the Postprocessor system and re-factors the Eigen based objects to avoid using FEProblem parameters.

### CODE REVIEW

File: framework/src/kernels/EigenKernel.C [hunk 1]
```diff
@@ -57,7 +80,7 @@ EigenKernel::computeResidual()
   _local_re.zero();
 
   Real one_over_eigen = 1.0;
-  one_over_eigen /= *_eigenvalue;
+  one_over_eigen /= (*_eigenvalue);
```
Comment (jwpeterson) [inline framework/src/kernels/EigenKernel.C]: This is strange, why not just:
```
Real one_over_eigen = 1.0 / *_eigenvalue;
```
?

File: framework/src/kernels/EigenKernel.C [hunk 2]
```diff
@@ -82,7 +105,7 @@ EigenKernel::computeJacobian()
   _local_ke.zero();
 
   Real one_over_eigen = 1.0;
-  one_over_eigen /= *_eigenvalue;
+  one_over_eigen /= (*_eigenvalue);
```
Comment (jwpeterson) [inline framework/src/kernels/EigenKernel.C]: Same thing here.

**Comment by jwpeterson** [comment]
Talked to @aeslaughter, this will be handled through an integration branch.

**Comment by YaqiWang** [comment]
For an eigen problem like `Ax = 1/k Bx`, `k` can not be zero, right? If `k` is zero, we are in trouble, so I guess we can add a `mooseAssert()`.

**Pull request closed without merging**